client/scripts/controllers/app/recent_activity.ts

```diff
@@ -0,0 +1,57 @@
+import { OffchainThread, AddressInfo, OffchainComment, OffchainReaction } from 'models';
+import RecentActivityStore, { IKeyedAddressCountAndInfo, IAddressCountAndInfo } from '../../stores/ActivityStore';
+class RecentActivityController {
+  private _store = new RecentActivityStore();
+
+  public get store() { return this._store; }
+
+  private _initialized = false;
+
+  public get initialized() { return this._initialized; }
+
+  public addThreads(threads: OffchainThread[]) {
+    threads.forEach((thread) => this._store.addThread(thread));
+  }
+
+  public addAddresses(addresses: AddressInfo[], community: string) {
+    addresses.forEach((addr) => this._store.addAddress(addr, community));
+  }
+
+  public addAddressesFromActivity(activity: any[]) {
+    activity.forEach((item) => {
+      const parentEntity = item.community || item.chain;
+      if (parentEntity === 'internal') console.log(item);
+      this._store.addAddress(item.Address, parentEntity);
+    });
+  }
+
+  public removeAddressActivity(activity: any[]) {
+    activity.forEach((item) => {
+      const parentEntity = item.community || item.chain;
+      const addressId = item.Address?.id || item.address_id || item.author;
+      this._store.removeAddressActivity(addressId, parentEntity);
+    });
+  }
+
+  public getThreadsByCommunity(community: string): Array<OffchainThread> {
+    return this._store.getThreadsByCommunity(community);
+  }
+
+  public getAddressesByCommunity(community: string): Array<AddressInfo> {
+    return this._store.getAddressesByCommunity(community);
+  }
+
+  public getAddressActivityByCommunity(community:string): IKeyedAddressCountAndInfo {
```

Comment:
this typing could be expressed as { [string]: IAddressCountAndInfo }, is there a reason we want to call it IKeyedAddressCountAndInfo? i feel like the name doesn't add much meaning. ICommunityAddressCountAndInfos would also make more sense, though it's a little hard to read

(i don't actually have a strong opinion on this, feel free to keep it as is too)

Reply:
Well, the `AddressCountAndInfo` differs from `IKeyedAddressCountAndInfo` insofar as it is "keyed" to the address ID. I guess I could call it `IKeyScopedAddressCountAndInfo`. But the difference is that it's an object with IDs as keys and the existing `AddressCountAndInfo` interface as values.

Reply:
It is hard to read though... I'll think about more eloquent names for these interfaces.